servant/src/Servant/API/BoundedNat.hs

```diff
@@ -0,0 +1,69 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+
+module Servant.API.BoundedNat (BoundedNat (unBoundedNat), unsafeBoundedNat) where
+
+import           Control.Monad    (unless)
+import           Data.Aeson
+import           Data.Aeson.Types (modifyFailure)
+import           Data.Bifunctor   (bimap)
+import           Data.Ix
+import           Data.Proxy       (Proxy (Proxy))
+import qualified Data.Text        as T
+import           GHC.Generics     (Generic)
+import           GHC.TypeLits
+import           Servant.API
+
+-- | A newtype wrapper around 'Word' that ensures the value is within a given range.
+--
+-- Example:
+--
+-- >>> :set -XDataKinds
+-- >>> import Data.Proxy
+-- >>> let val = unsafeBoundedNat 5 :: BoundedNat 1 10
+-- >>> unBoundedNat val
+-- 5
+-- >>> let invalidVal = unsafeBoundedNat 15 :: BoundedNat 1 10
+-- >>> unBoundedNat invalidVal
+-- 15
+-- >>> checkBoundedRange (const $ fail "Out of range") val
```

Comment:
I'm quite intrigued by this workflow. Wouldn't it be easier to make a smart creator (parser) that returns a `Maybe (Range n1 n2)` instead of having a validation function called after the fact?

Reply:
I think I can try and refactor to apply this suggestion, thanks for the feedback! 👍🏻